pom.xml

```diff
@@ -648,13 +648,22 @@
                         <eclipse>
                             <file>style/spotless_dolphinscheduler_formatter.xml</file>
                         </eclipse>
+                        <removeUnusedImports />
+                        <importOrder>
+                            <file>style/eclipse.importorder</file>
+                        </importOrder>
+                        <replaceRegex>
```

Comment:
@kezhenxu94 https://github.com/diffplug/spotless/issues/649#issuecomment-662739583 It is hardly possible for `Spotless` to replace with `wildcards` with `specific imports`, thus this is some kind of workaround actually. To prevent developers from pushing the commits without knowing that `wildcards` has been removed by their pre-commit hook, one possible solution is to add a message in pre-commit hook to inform them of this behavior. WDYT?

Reply:
FYI, I also added a comment in the origin issue to seek help: https://github.com/diffplug/spotless/issues/649#issuecomment-1213743932

Reply:
> @kezhenxu94 FYI, I also added a comment in the origin issue to seek help to see if there is some good practice to handle `wildcard imports` with `Spotless`: https://github.com/diffplug/spotless/issues/649#issuecomment-1213743932

Wow. The issue has been 2 years...

Reply:
> This removes the star import. But will it instead replaces the star import with the used specific class import? If not, this will cause compiling failed. Can you change a class with this example?

@kezhenxu94 Just a follow-up, as I've tested it locally, if we use `mvn spotless:check` instead of `mvn spotless:apply`, it only blocks the `wildcard imports` without removing them. To improve, we could change the pre-commit hook from `mvn spotless:apply` to `mvn spotless:check` and add an output message to inform contributors that they should remove `wildcard imports`. WDYT?

Reply:
> > This removes the star import. But will it instead replaces the star import with the used specific class import? If not, this will cause compiling failed. Can you change a class with this example?
> 
> @kezhenxu94 Just a follow-up, as I've tested it locally, if we use `mvn spotless:check` instead of `mvn spotless:apply`, it only blocks the `wildcard imports` without removing them. To improve, we could change the pre-commit hook from `mvn spotless:apply` to `mvn spotless:check` and add an output message to inform contributors that they should remove `wildcard imports`. WDYT?

Can you identify the failure of `spotless:check` is because of `wildcard imports` or other code style issues?

Reply:
@kezhenxu94 Yes, I just double-checked it. To reproduce, you might change https://github.com/apache/dolphinscheduler/blob/68316f68fa9b90560846c3a12274edec2bf5d960/dolphinscheduler-task-plugin/dolphinscheduler-task-zeppelin/src/main/java/org/apache/dolphinscheduler/plugin/task/zeppelin/ZeppelinTask.java#L33-L35 to `import java.util.*;` and run `mvn spotless:check`.

Reply:
> @kezhenxu94 Yes, I just double-checked it. To reproduce, you might change
> 
> https://github.com/apache/dolphinscheduler/blob/68316f68fa9b90560846c3a12274edec2bf5d960/dolphinscheduler-task-plugin/dolphinscheduler-task-zeppelin/src/main/java/org/apache/dolphinscheduler/plugin/task/zeppelin/ZeppelinTask.java#L33-L35
> 
> to `import java.util.*;` and run `mvn spotless:check`.

How can yo tell the `spotless:check` failure is because of wildcard import, if I have other code style issues and no wildcard import?

Reply:
> > @kezhenxu94 Yes, I just double-checked it. To reproduce, you might change
> > https://github.com/apache/dolphinscheduler/blob/68316f68fa9b90560846c3a12274edec2bf5d960/dolphinscheduler-task-plugin/dolphinscheduler-task-zeppelin/src/main/java/org/apache/dolphinscheduler/plugin/task/zeppelin/ZeppelinTask.java#L33-L35
> > 
> > to `import java.util.*;` and run `mvn spotless:check`.
> 
> How can yo tell the `spotless:check` failure is because of wildcard import, if I have other code style issues and no wildcard import?

Good question, since each rule has its name, I think there might be a way to configure `Spotless` to explicitly tell which rule the code fails. I will take a look and see if there is a way to do so. 
![image](https://user-images.githubusercontent.com/34905992/184781301-acc0b19c-ccdf-442e-a5a3-08eb8f7705a5.png)